firmware/thruna/PDM/test/test_stateMachine.cpp

```diff
@@ -0,0 +1,20 @@
+#include "test_pdmBaseStateMachineTest.h"
+
+class StateMachineTest : public PdmBaseStateMachineTest
+{
+};
+
+TEST_F(StateMachineTest, check_init_state_is_broadcasted_over_can)
+{
+    SetInitialState(app_initState_get());
+    LetTimePass(state_machine, 10);
+    EXPECT_EQ(PDM_INIT_STATE, App_CanTx_PDM_State_Get());
+}
+
+TEST_F(StateMachineTest, check_state_transition_from_init_to_drive)
+{
+    SetInitialState(app_initState_get());
+    App_CanRx_BMS_State_Update(BMS_INVERTER_ON_STATE);
+    LetTimePass(state_machine, 10);
+    EXPECT_EQ(PDM_DRIVE_STATE, App_CanTx_PDM_State_Get());
+}
```

Comment:
These tests could be a bit more robust. 

1. There is no check that the PDM remains in its intended state, it could immediately transition to another state and the tests wouldn't catch that. A simple way to check this is just wait a large amount of time (ex. 1s) and then check that it is still in the intended state.
2. For the second test, you can't say for sure that it was your action of updating the BMS RX message that caused the state transition. Theoretically, it could have done that by itself. A way to make sure is again to just wait a large amount of time before setting your test condition, and then you can have more confidence it was actually you causing the transition.
3. You don't test that the power manager also changes its state. Maybe add a `app_powerManager_getState` function and use that to check the PDM is in the correct power state at all times.
4. We don't test the DRIVE -> INIT condition.

I'm being pedantic in this case, but in general I think these are good things to think about when writing tests so I thought I'd mention them.